Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Promact.Trappist.DomainModel.Models.Category;
+using Promact.Trappist.Repository.Categories;
+
+namespace Promact.Trappist.Core.Controllers
+{
+    [Route("api/category")]
+    public class CategoryController : Controller
+    {
+        private readonly ICategoryRepository _categoryRepository;
+        public CategoryController(ICategoryRepository categoryRepository)
+        {
+            _categoryRepository = categoryRepository;
+        }
+        /// <summary>
+        /// Get All Categories
+        /// </summary>
+        /// <returns>Category List</returns>
+        [HttpGet]
+        public IActionResult GetAllCategories()
+        {
+            var categoryList = _categoryRepository.GetAllCategories();
+            return Ok(categoryList);
+        }
+
+        #region post Method 
+        [HttpPost]
+        /// <summary>
+        /// Post Method 
+        /// Will Add a Catagory Name in Category Model
+        ///</summary>
+        /// <param name="category">Object of  class Category</param>
+        /// <returns>object of the class </returns>
+        public IActionResult catagoryAdd([FromBody] Category category)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+            _categoryRepository.AddCategory(category);
+            return Ok(category);
+        }
+        #endregion
+
+        #region PutMethod
+        /// <summary>
+        /// Put Method
+        /// Will Edit a Existing Category from Category Model
+        /// </summary>
+        /// <param name="catagory">Object of  class Category</param>
+        /// <returns>object of the class after key found </returns>
+        [HttpPut("{id}")]
+        public IActionResult CategoryEdit([FromRoute] int id, [FromBody] Category category)
+        {
+            var categoryName = category.CategoryName;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused 'categoryName' local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1481)